pkg/pool-linear/contracts/aave/AaveLinearPoolFactory.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+import "@balancer-labs/v2-pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol";
+import "@balancer-labs/v2-pool-utils/contracts/factories/FactoryWidePauseWindow.sol";
+
+import "./AaveLinearPool.sol";
+
+contract AaveLinearPoolFactory is BasePoolSplitCodeFactory, FactoryWidePauseWindow {
```

Comment:
I'd remove `MockLinearPoolFactory` and change its tests so that it tests this factory instead of the mock one. Those tests are important as they ensure that the `_create` call is correct, which is doubly critical now that we have the manual `abi.encode` of the raw arguments instead of having the constructor struct (which gave us some type safety).

Once we develop a second type of LinearPool and a corresponding second factory we can think about whether we want to share code among factories or how to reuse the tests, but that's not something we need to worry about now.